Engine/Module/ModuleResourceManager.h

```diff
@@ -43,6 +49,19 @@ class TextureImporter;
 
 class SceneManager;
 
+struct TextureLoadJob
+{
+	uint32_t uuid = 0;
+	bool already_in_cache = false;
+	FileData exported_file_data;
+	Component* component_to_load = nullptr;
+	TextureLoadData loaded_data;
```

Comment:
Why are you retrieving the loaded data directly instead of the uuid? I mean you created a method for the componentes for that passing the loaded data, and then you need to add it to the cache manually etc, wouldn't be easier if you just save de uuid and then call the already existing method load?

Reply:
I do it as you told me, in two steps (main thread and off thread). First when the normal "load" is called and is a texture I save its uuid (inside of textureloadeddata), component and exported_file_data. Then I add this to the loading_queue. And the offthread takes this information and reads the exported_file_data (the Load of the TextureManager that is what takes more time to compute) and copy the info (data, height, width, num_channels, uuid, texture_options) into a new loaded data that we push to the processing_queue (the one that reads and treats the main thread. That create the Texture resource with the LoadedData (doing all the OpenGL calls). The thing is what I want to do offthread is the Load of the TextureManager that it takes 5 seconds on mainthread. Thats why I don't use the normal Load. 
I don't know if this answers your question. If not DM me on dsc :D